flow/aggregates.py

```diff
@@ -405,12 +396,10 @@ def __len__(self):
         return len(self._project)
 
     def __eq__(self, other):
-        return type(self) == type(other) and \
-               self._project == other._project
+        return type(self) == type(other) and self._project == other._project
 
     def __hash__(self):
-        blob = self._project.get_id()
-        return int(_hash(blob), 16)
+        return hash(repr(self._project))
```

Comment:
**This was not a safe hash function.** The project id is just a name, and isn't actually a unique identifier (two projects in different locations can have the same name, and thus the same id). For the Project's `__eq__` method, we use `repr(self) == repr(other)`, because the `repr` includes a unique identifier: the project's path on disk. I use the same approach here.